### PR TITLE
[19.03 backport] Update DNS library

### PR DESCRIPTION
**Pull request by SamWhited**
This makes sure that we don't become vulnerable to CVE-2018-17419 or
CVE-2019-19794 in the future.

This is a cherry-pick  of #40579 using the correct `bump_19.03` libnetwork branch.

Signed-off-by: Sam Whited <sam@samwhited.com>

### CODE REVIEW

**Comment by thaJeztah** [comment]
Don't merge this one yet; looks like there's a regression; https://github.com/moby/moby/issues/40634#issuecomment-595824120

File: vendor/github.com/docker/libnetwork/vendor.conf [hunk 1]
```diff
@@ -34,7 +34,7 @@ github.com/sean-/seed                   e2103e2c35297fb7e17febb81e49b312087a2372
 github.com/hashicorp/go-sockaddr        c7188e74f6acae5a989bdc959aa779f8b9f42faf # v1.0.2
 github.com/hashicorp/serf               598c54895cc5a7b1a24a398d635e8c0ea0959870
 github.com/mattn/go-shellwords          02e3cf038dcea8290e44424da473dd12be796a8a # v1.0.3
-github.com/miekg/dns                    e57bf427e68187a27e22adceac868350d7a7079b # v1.0.7
+github.com/miekg/dns                    6c0c4e6581f8e173cc562c8b3363ab984e4ae071 # v1.1.27
```
Comment (thaJeztah) [inline vendor/github.com/docker/libnetwork/vendor.conf]: Looks like you forgot to update this in the local `vendor.conf`

Comment (SamWhited) [reply vendor/github.com/docker/libnetwork/vendor.conf]: Oops, unsure how that happened. Fixed.

**Comment by trapier** [comment]
@thaJeztah: hi!

Looks like this is up to date on the action items from https://github.com/moby/moby/pull/40596#issuecomment-596624901.  Should be good to go.

**Comment by thaJeztah** [comment]
looks like there's a merge conflict; @samwhited can you rebase and remove the temporary commit?

**Comment by SamWhited** [comment]
@thaJeztah Rebased, tests running.

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by thaJeztah** [comment]
could you update the commit once more to include https://github.com/moby/libnetwork/pull/2541 ?

**Comment by SamWhited** [comment]
Done

**Review by thaJeztah — APPROVED** [review]
still LGTM

**Comment by SamWhited** [comment]
Test failures appear to exist in master too.

**Comment by thaJeztah** [comment]
Networking in CI has quite some issues recently. No idea what changed tbh

**Comment by SamWhited** [comment]
@thaJeztah can this be merged or do we need to figure out the CI flakiness first?

**Comment by euanh** [comment]
I see some evidence of network connectivity problems from the build worker in the build logs:

1. ppc64le: failed to clone containerd, `GnuTLS recv error`; s390x: failed apt update, `Temporary failure resolving 'cdn-fastly.deb.debian.org'`
2. s390: failed apt update `Temporary failure resolving 'cdn-fastly.deb.debian.org'`
3. win_RS5: not sure, could be a Git clone failure
4. amd64: test failure `amd64.integration-cli TestDockerSwarmSuite/TestSwarmClusterRotateUnlockKey`
5. s390: failed apt update `Temporary failure resolving 'cdn-fastly.deb.debian.org'`
6. win_RS5: timeout?
7. PASS
8. superseded by 9`
9. amd64: error pulling an asset during swagger build `error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function`
10. amd64: error in frozen-images `curl: (56) OpenSSL SSL_read: SSL_ERROR_SYSCALL, errno 101`
11. amd64: criu build failed pulling from the network - asset was corrupted: `curl -sSL https://github.com/checkpoint-restore/criu/archive/v${CRIU_VERSION}.tar.gz | tar -C /usr/src/criu/ -xz --strip-components=1` -> `gzip: stdin: not in gzip format`

**Comment by euanh** [comment]
@thaJeztah I think it's at least worth kicking off the build again.  The last runs were a week ago and if there were networking problems for the builder they may have been resolved.

**Comment by SamWhited** [comment]
Reran CI, it's green now.

**Comment by euanh** [comment]
:tada:

**Comment by thaJeztah** [comment]
@cpuguy83 @AkihiroSuda good to go?

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**